File: Baekjoon4948.py
from math import *

def prime_num(x) :
    if x == 1 :
        return False
    for i in range(2, int(pow(x,0.5)+1)) :
        if(x % i == 0) :
            return False
    return True

all_list = list(range(2,246912))
memo = []

for i in all_list:
    if prime_num(i) :
        memo.append(i)

num = int(input())

while 1 :
    cnt = 0
    if num == 0 :
        break

    for i in memo :
        if(num < i <= num * 2) :
            cnt += 1
    
    print(cnt)
    num = int(input())

File: Baekjoon4948.c
//
// Created by 하윤 on 2021/08/22.
// prob.4948
#include <stdio.h>
#include <math.h>

int main(void) {
    int num;
    int prime[246913] = {0, };
    prime[0] = 1, prime[1] = 1;
    for(int i = 2; i < 246913 / i; i++) {
        //if(prime[i] == 1) continue;
        for(int j = i * i; j < 246913; j += i) {
            if(j % i == 0)
                prime[j] = 1;
        }
    } int cnt;

    while(1) {
        scanf("%d", &num);
        if(num == 0)
            break;
        cnt = 0;
        for(int i = num + 1; i <= num * 2; i++) {
            if (prime[i] == 0)
                cnt++;
        }
        printf("%d\n",cnt);

    }
    return 0;
}

